core/src/main/java/tc/oc/pgm/loot/FillerDefinition.java

```diff
@@ -0,0 +1,94 @@
+package tc.oc.pgm.loot;
+
+import java.time.Duration;
+import java.util.Objects;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.feature.FeatureDefinition;
+import tc.oc.pgm.api.feature.FeatureInfo;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.util.compose.Composition;
+
+@FeatureInfo(name = "fill")
+public class FillerDefinition implements FeatureDefinition {
+
+  private final Composition<ItemStack> loot;
+  private final Filter filter;
+  private final @Nullable Filter refillTrigger; // Dynamic
```

Comment:
Would this being null be functionally equivalent to simply an always (or a never) filter?
if so, may as well make it non-null and use an always/never for it

Reply:
If this is null it never registers a dynamic filter listener(See LootableMatchModule constructor), so to prevent unnecessary registering you would have to do an `#equal` check instead.